core/src/main/java/info/openrocket/core/simulation/RK6SimulationStepper.java

```diff
@@ -0,0 +1,656 @@
+import info.openrocket.core.simulation.listeners.SimulationListenerHelper;
+import info.openrocket.core.startup.Application;
+import info.openrocket.core.util.Coordinate;
+import info.openrocket.core.util.MathUtil;
+import info.openrocket.core.util.Quaternion;
+import info.openrocket.core.util.WorldCoordinate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Random;
+
+
+public class RK6SimulationStepper extends AbstractSimulationStepper {
+
+    private static final Logger log = LoggerFactory.getLogger(RK6SimulationStepper.class);
+    private static final Translator trans = Application.getTranslator();
+
+
+    /** Random value with which to XOR the random seed value */
+    private static final int SEED_RANDOMIZATION = 0x23E3A01F;
+
+
+    /**
+     * A recommended reasonably accurate time step.
+     */
+    public static final double RECOMMENDED_TIME_STEP = 0.05;
+
+    /**
+     * A recommended reasonable maximum simulation time (in seconds).
+     */
+    public static final double RECOMMENDED_MAX_TIME = 1200;
+
+    /**
+     * A recommended maximum angle step value.
+     */
+    public static final double RECOMMENDED_ANGLE_STEP = 3 * Math.PI / 180;
+
+    /**
+     * A random amount that is added to pitch and yaw coefficients, plus or minus.
+     */
+    public static final double PITCH_YAW_RANDOM = 0.0005;
+
+    /**
+     * Maximum roll step allowed.  This is selected as an uneven division of the full
+     * circle so that the simulation will sample the most wind directions
+     */
+    private static final double MAX_ROLL_STEP_ANGLE = 2 * 28.32 * Math.PI / 180;
+    //	private static final double MAX_ROLL_STEP_ANGLE = 8.32 * Math.PI/180;
+
+    private static final double MAX_ROLL_RATE_CHANGE = 2 * Math.PI / 180;
+    private static final double MAX_PITCH_YAW_CHANGE = 4 * Math.PI / 180;
+
+    private Random random;
+    DataStore store = new DataStore();
+
+    @Override
+    public SimulationStatus initialize(SimulationStatus original) {
+
+        SimulationStatus status = new SimulationStatus(original);
+        // Copy the existing warnings
+        status.setWarnings(original.getWarnings());
+
+        SimulationConditions sim = original.getSimulationConditions();
+
+        store.launchRodDirection = new Coordinate(
+                Math.sin(sim.getLaunchRodAngle()) * Math.cos(Math.PI / 2.0 - sim.getLaunchRodDirection()),
+                Math.sin(sim.getLaunchRodAngle()) * Math.sin(Math.PI / 2.0 - sim.getLaunchRodDirection()),
+                Math.cos(sim.getLaunchRodAngle()));
+
+        this.random = new Random(original.getSimulationConditions().getRandomSeed() ^ SEED_RANDOMIZATION);
+
+        return status;
+    }
+
+
+    @Override
+    public void step(SimulationStatus status, double maxTimeStep) throws SimulationException {
+
+        status.storeData();
+
+        ////////  Perform RK4 integration:  ////////
```

Comment:
Should this be RK6?

Reply:
Should be all set in `f4d47579d68af4c9350bacf9f3e73703ea254a91`.